ArduPlane/mode.cpp

```diff
@@ -201,7 +201,22 @@ void Mode::update_target_altitude()
     } else if (plane.target_altitude.offset_cm != 0 && 
                !plane.current_loc.past_interval_finish_line(plane.prev_WP_loc, plane.next_WP_loc)) {
         // control climb/descent rate
+#if HAL_QUADPLANE_ENABLED
+        // On climbout we will give the ability to do a full rate climb after the VTOL Takeoff. We
+        // will set the glide slope proportion to the destination to achieve this.
+        if (quadplane.get_do_full_rate_climbout_enable() && quadplane.get_do_full_rate_climbout()) {
+            // Set us to the completed portion to get full rate climb. Here, 0 is completed, see
+            // the docstring for set_target_altitude_proportion.
+            plane.set_target_altitude_proportion(plane.next_WP_loc, 0.0f);
```

Comment:
I'm surprised this isn't the same as the Auto code, which does a `reset_offset_altitude`